examples/hosts/node-host/src/nodeCodeloader.ts

```diff
@@ -50,26 +51,38 @@ export class NodeCodeLoader {
         const codeEntrypoint = `${packageDirectory}/node_modules/${scope}/${name}`;
 
         // Install the versioned package if not already present.
-        if (!fs.existsSync(packageDirectory)) {
-            fs.mkdirSync(packageDirectory, { recursive: true });
-
-            // Copy over the root .npmrc (if present) to the directory where npm install will be executed.
-            if (fs.existsSync(`${this.packageDirectory}/.npmrc`)) {
-                fs.copyFileSync(`${this.packageDirectory}/.npmrc`, `${packageDirectory}/.npmrc`);
-            }
-
-            // Run npm install
-            await asyncExec(`npm install ${pkg}`, { cwd: packageDirectory });
+        if (!fs.existsSync(signalPath) || !fs.existsSync(codeEntrypoint)) {
+            const pkgDirExists = fs.existsSync(packageDirectory);
+            if (!pkgDirExists || this.useLocalDirectory) {
```

Comment:
The behavior is same even if useLocalDirectory is true. Consider you run two instance in your machine. useLocalDirectory will only dictate whether you should use __dirname or not. Otherwise the behavior is same.